xmppserver/src/main/java/org/jivesoftware/openfire/net/SocketReader.java

```diff
@@ -366,6 +367,18 @@ private IQ getIQ(Element doc) {
         Element query = doc.element("query");
         if (query != null && "jabber:iq:roster".equals(query.getNamespaceURI())) {
             return new Roster(doc);
+        }else if (query != null && "jabber:iq:version".equals(query.getNamespaceURI())) {
```

Comment:
I think (but have not tested) that this has an unexpected side-effect. It will not only process the software version response from the server (you might want to add a check to only process result stanzas, you're currently also processing 'request' stanzas), but you will also process the software version responses from any user on that server.

I suspect that, when user X on our server asks for the software version of the client of user Y on the other server, this bit of code will cause the software version of the other server to be reported as the client of user Y. That's probably not good. :)

Please test for this, and make changes if needed.

Reply:
> I suspect that, when user X on our server asks for the software version of the client of user Y on the other server, this bit of code will cause the software version of the other server to be reported as the client of user Y. That's probably not good. :)

Yes, I tested and had that side-effect, the software version of the other server has reported as the client of the user :)
So what Should I do? maybe SocketReader it's not a good place for that code

Reply:
I think this is a good place for that code. You should, however, make the code smarter: it must check if the sender of the version result is the remote server (`from="example.org"`) and not a user on the remote server (`from="userY@example.org"`) or any other entity on that server (from="`mycomponent.example.org`").

Reply:
Your update is a good improvement, but still has one issue: the answer of an external component (`component.example.org`) on the remote server `example.org` would still be interpreted as the response of the server itself.

Reply:
I think that the current state of the code still has this problem: When an external component on a remote server responds with Software Version information, then Openfire, based on this code, probably will record the software version as being reported by the remote server, instead of the remote external component.